src/test/scala/tech/cryptonomic/conseil/util/ConfigUtilTest.scala

```diff
@@ -187,13 +188,53 @@ class ConfigUtilTest extends WordSpec with Matchers with EitherValues {
         platforms.values.flatten should contain only (
           TezosConfiguration(
             "alphanet",
-            TezosNodeConfiguration(hostname = "localhost", port = 8732, protocol = "http")
+            TezosNodeConfiguration(hostname = "localhost", port = 8732, protocol = "http"),
+            None
           ),
           UnknownPlatformConfiguration("some-network")
         )
 
       }
 
+      "extract a configuration map that includes a known TNS contract" in {
+        import Platforms._
+        import tech.cryptonomic.conseil.util.ConfigUtil.Pureconfig._
+
+        val cfg = ConfigFactory.parseString("""
+          | platforms.tezos : {
+          |  alphanet: {
+          |    node: {
+          |      protocol: "http",
+          |      hostname: "localhost",
+          |      port: 8732
+          |      pathPrefix: ""
+          |    },
+          |    tns: {
```

Comment:
TNS is not an intrinsic feature of the blockchain so I don't think it should belong with the node configuration. IMHO, It should have an entry of its own within the Lorre conf.

Reply:
This configuration is very network-specific since it needs to identify the contract for each net.
Hence I assumed that you might want to have different configurations, or none at all (fallbacking to a no-op), for each network configuration.
I used this locally to make tests, so that I keep a single configuration file with multiple networks available and can easily run Lorre on a different one, via a single CLI argument.
Otherwise in my case, I should rewrite the `tns` value every time I swap net, keeping commented out copies.

I don't see this entry as related to the node, as much as to the network used, alongside the specific node coordinates.

Anyway, I can bring it at top-level, if we expect every user to have only a single network configured per conf file.

Reply:
I think there's a happy compromise. We can bring TNS to the top levels and allow per-network specification of contracts. What do you think?

Reply:
Sounds fine but let me check I did understand:

we have a root level `tns` config object, and inside we can put different config entries, each with the key being the network, like this

```
tns {
  mainnet { ... }
  carthagenet { ... }
}
```

This object definition would be empty in the `reference.conf`.

Do I get this right?

Reply:
Yep, that’s what I meant!

Reply:
Great, on it! 👍

Reply:
done